### PR TITLE
For #110: Added meaningful error for broken package in Rpm.batchUpdate

### PR DESCRIPTION
**Pull request by paulodamaso**
For #110:
- added test for checking if `Rpm.batchUpdate` is throwing an exception when it has a broken package.
- minor typo correction
- left puzzle for implementing the behavior

@g4s8 Notice that the behavior somewhat conflicts with https://github.com/artipie/rpm-adapter/issues/190 : if we are expecting for an exception to be thrown, the metadata content won't be changed after all. Also seems that we should add packages through the `Rpm` interface instead adding them directly to storage, and call `batchUpdate` internally

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#210](https://github.com/artipie/rpm-adapter/issues/210) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/fcc6460c-87df-466b-8142-e2f4b0ba99ab); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/rpm/RpmTest.java [hunk 1]
```diff
@@ -88,4 +93,30 @@ void doesntBrakeMetadataWhenInvalidPackageSent(@TempDir final Path tmp)
...
+        storage.save(
+            new Key.From("brokentwo.rpm"),
+            new Content.From(
+                broken
+            )
+        );
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
+            () -> repo.batchUpdate(Key.ROOT).blockingAwait(),
+            "RPM package \"brokentwo.rpm\" is broken, can't be read"
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/rpm/RpmTest.java:119]: @paulodamaso let's rephrase this a bit:

> Reading of RPM package <package> failed, data corrupt or malformed.

**Comment by paulodamaso** [comment]
@carlosmiranda Done, please take a look

File: src/test/java/com/artipie/rpm/RpmTest.java [hunk 2]
```diff
@@ -89,4 +94,30 @@ void doesntBrakeMetadataWhenInvalidPackageSent(@TempDir final Path tmp)
...
+        final Rpm repo =  new Rpm(
+            storage, StandardNamingPolicy.SHA1, Digest.SHA256, true
+        );
+        storage.save(
+            new Key.From("stored.rpm"),
+            new Content.From("stored content".getBytes())
+        );
+        repo.batchUpdate(Key.ROOT).blockingAwait();
+        final byte[] broken = {0x00, 0x01, 0x02 };
+        storage.save(
```
Comment (g4s8) [inline src/test/java/com/artipie/rpm/RpmTest.java:111]: @paulodamaso it's async call which returns `CompletableFuture`, you either need to wait it with `.toCompletableFuture().get()` or wrap this call with `BlockingStorage`

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@g4s8 Corrections made, please take a look

**Comment by olenagerasimova** [comment]
@paulodamaso thanks for the corrections and comment, I want to discuss one moment though: as you correctly noticed, #190 (closed already) and #110 have some  behavior conflicts. In #110 we decided to that failing the update is more correct, which means, that #110 is not finished yet, but I do not see any puzzles in this PR. Please, could you add the puzzle and explain what we should fail the update and do not touch metadata if invalid package was met?

**Comment by paulodamaso** [comment]
@olenagerasimova Done, please take a look

**Comment by olenagerasimova** [comment]
@paulodamaso please resolve conflict

**Comment by olenagerasimova** [comment]
@paulodamaso please resolve conflicts

**Comment by paulodamaso** [comment]
@olenagerasimova Done, please merge

**Comment by olenagerasimova** [comment]
@g4s8 please approve and I'll merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (17 days), architects (@g4s8, @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by sereshqua** [comment]
@carlosmiranda please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by carlosmiranda** [comment]
@sereshqua confirmed, sorry for the delay. I suffered a laptop crash last week.

**Comment by sereshqua** [comment]
@carlosmiranda np :)

**Pull request merged**